doc/developer/design/20230714_show_privileges.md

```diff
@@ -0,0 +1,202 @@
+- [System privilege design doc](20230531_system_privileges.md)
+- [Default privilege design doc](20230602_default_privileges.md)
+
+## Context
+
+Role based access control comprises a set of features that allow customers to control the
+privileges of individual users. Privileges can be granted to a user which allows them to take
+certain actions in a Materialize deployment. All objects have an owner, and certain actions are
+limited to only object owners. Roles can be members of other roles and inherit all of their
+privileges. Privileges, owners, and role membership are all stored within the database catalog.
+
+Currently, it's difficult for a user to discover RBAC related information.
+
+## Goals
+
+- Make it easy for users to discover the set of privileges that they have.
+- Make it easy for users to discover the owner of objects.
+- Make it easy for users to discover the roles that they are a member of.
+
+## Overview
+
+We will add the following SQL commands:
+
+- `SHOW PRIVILEGES`: shows the privileges of the current user.
+- `SHOW DEFAULT PRIVILEGES`: shows default privileges.
+- `SHOW ROLE MEMBERSHIP`
+
+We will also add an `owner` column to the existing show commands.
+
+## Detailed description
+
+- We will add an internal unmaterializable function, `mz_internal.mz_role_membership() -> text[]`,
+  that returns an array of all the role IDs that the session's current role is a member of.
+- We will add an internal unary function, `mz_internal.mz_format_privileges(text) -> text`,
+  that accepts a privilege string descriptor and returns a human-readable version. For
+  example: `mz_internal.mz_format_privileges('ar') = 'INSERT, SELECT'`.
+
+These two functions will help implement the `SHOW` commands described below.
+
+### `SHOW PRIVILEGES`
+
+Will show all privileges held by the current session.
+
+The syntax will be: `SHOW PRIVILEGES [ON <object-type>] [FROM <schema-name>]`.
+
+- `ON <object-type>` will filter the output to only include objects of type `<object-type>`. Valid
+  values are:
+    - `TABLES`
+    - `VIEWS`
+    - `MATERIALIZED VIEWS`
+    - `SOURCES`
+    - `SINKS`
+    - `TYPES`
+    - `CLUSTERS`
+    - `SECRETS`
+    - `CONNECTIONS`
+    - `DATABASES`
+    - `SCHEMAS`
+    - `SYSTEM`
+- `FROM <schema-name>` will filter the output to only exclude objects in a schema other `<schema-name>`.
+    - If specified, top level objects, such as database, schemas, and clusters are not included.
+    - If not specified then objects are limited to the current schema and top level objects are
+    - shown.
+
+The output will include the following columns:
+
+- `name: text` The name of the object.
+- `object_type: text` The type of the object.
+- `privileges: text` Human readable version of the privileges that held by the current session.
+
+Here are some example queries:
+
+```sql
+SHOW PRIVILEGES;
+
+ name       | object_type | privileges
+------------+-------------+----------------
+ my_table   | TABLE       | SELECT, INSERT
```

Comment:
The object type should be lowercase, IIRC, to match how it's rendered in `mz_objects`.

```suggestion
 my_table   | table       | SELECT, INSERT
```

Reply:
In that case, we should probably also fix the casing in `mz_default_privileges`

```
materialize=> SELECT * FROM mz_default_privileges;
 role_id | database_id | schema_id | object_type | grantee | privileges 
---------+-------------+-----------+-------------+---------+------------
 p       |             |           | TYPE        | p       | U
 p       |             |           | SCHEMA      | s2      | U
 p       |             |           | CLUSTER     | s2      | U
 p       |             |           | DATABASE    | s2      | U
(4 rows)

```

Reply:
I updated the casing in this doc.